### PR TITLE
Migrate from raven-js to @sentry/browser

### PR DESCRIPTION
**Pull request by robertknight**
Migrate frontend crash reporting from Sentry's old raven-js library to the modern `@sentry/browser` JS library.

**Testing:**

1. Copy the DSN value from https://hypothesis.sentry.io/settings/projects/frontend-h/keys/
2. Set the `SENTRY_DSN_FRONTEND` variable in h's environment to the DSN value
3. Load a page in h, you should see a request to https://o18985.ingest.us.sentry.io/api/146908/envelope/

### CODE REVIEW

File: h/static/scripts/base/sentry.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/**
```
Comment (robertknight) [inline h/static/scripts/base/sentry.ts:1]: This has the same structure as the old `raven.js` module, but converted to TS and adapted to use the new Sentry API. I also removed the unhandled-promise code as that is obsolete.

File: h/static/scripts/tests/base/sentry-test.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import * as sentry from '../../base/sentry';
```
Comment (robertknight) [inline h/static/scripts/tests/base/sentry-test.js:1]: This is `raven-test.js` but updated to match the new APIs.

File: rollup-tests.config.js [hunk 3]
```diff
@@ -13,6 +13,7 @@ export default {
   plugins: [
     nodeResolve({
       browser: true,
+      extensions: ['.js', '.ts', '.tsx'],
```
Comment (robertknight) [inline rollup-tests.config.js:16]: This allows JS unit tests to import TS modules.

File: h/static/scripts/base/sentry.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    //
+    // If the error is a plain object or non-Error subclass with a message
+    // property, such as errors returned by chrome.extension.lastError,
+    // use that instead.
+    if (typeof error === 'object' && error && 'message' in error) {
+      error = error.message;
+    }
+  }
+
+  const extra = Object.assign({ when: when }, context);
```
Comment (acelaya) [inline h/static/scripts/base/sentry.ts:50]:
```suggestion
  const extra = Object.assign({ when }, context);
```

**Review by acelaya — COMMENTED** [review]
This worked for me, with the only consideration that the request to sentry results in a 403, but I assume that's expected and will work only from certain environments.

**Comment by robertknight** [comment]
Thanks @acelaya. I'll get this merged.

> This worked for me, with the only consideration that the request to sentry results in a 403, but I assume that's expected and will work only from certain environments.

If you used the production Sentry DSN, then this is indeed expected, as there is a filter configured in Sentry to limit which domains error reports are accepted from.

**Pull request merged**